src/Admin/Controllers/OrganizationsController.cs

```diff
@@ -208,13 +212,23 @@ public async Task<IActionResult> Edit(Guid id, OrganizationEditModel model)
             throw new BadRequestException("Plan does not support Secrets Manager");
         }
 
-        await _organizationRepository.ReplaceAsync(organization);
-        await _applicationCacheService.UpsertOrganizationAbilityAsync(organization);
-        await _referenceEventService.RaiseEventAsync(new ReferenceEvent(ReferenceEventType.OrganizationEditedByAdmin, organization, _currentContext)
+        try
         {
-            EventRaisedByUser = _userService.GetUserName(User),
-            SalesAssistedTrialStarted = model.SalesAssistedTrialStarted,
-        });
+            await _stripeSyncService.UpdateCustomerEmailAddress(organization.GatewayCustomerId, organization.BillingEmail);
+            await _organizationRepository.ReplaceAsync(organization);
+            await _applicationCacheService.UpsertOrganizationAbilityAsync(organization);
+            await _referenceEventService.RaiseEventAsync(new ReferenceEvent(ReferenceEventType.OrganizationEditedByAdmin, organization, _currentContext)
+            {
+                EventRaisedByUser = _userService.GetUserName(User),
+                SalesAssistedTrialStarted = model.SalesAssistedTrialStarted,
+            });
+        }
+        catch (StripeException stripeException)
+        {
```

Comment:
Don't love this by any stretch of the imagination, but the client is only using basic HTML 5 email validation, which something like "test@email" go through. I figured fixing the client-side validation was out of scope for this.

Reply:
⛏️ I think you should be fine to add validation for the email field if you're comfortable doing so, but no worries either way.

I'd suggest reducing the scope of the try/catch to just include line `217`. Since you're only catching `StripeException`, there's no danger that the existing code will trigger that exception, and since you're rethrowing the exception, execution will halt in the catch block preventing any further updates.